### PR TITLE
Update createBakeTask with manage image.

### PR DESCRIPTION
**Pull request by DanielaS12**
Add support for handling the bake of managed and custom images for Azure.

Tested the feature manually. 

To be pushed after the rosco packer templates have been validated and the entire workflow was tested.

### CODE REVIEW

File: orca-bakery/src/main/groovy/com/netflix/spinnaker/orca/bakery/tasks/CreateBakeTask.groovy [hunk 1]
```diff
@@ -188,10 +182,40 @@ class CreateBakeTask implements RetryableTask {
       requestMap.remove("baseName")
     }
 
+    if (stage.context.baseOs == null) {
```
Comment (ovidiupopa07) [inline orca-bakery/src/main/groovy/com/netflix/spinnaker/orca/bakery/tasks/CreateBakeTask.groovy:185]: can you please also add unit tests for the new functionality?

File: orca-bakery/src/main/groovy/com/netflix/spinnaker/orca/bakery/tasks/CreateBakeTask.groovy [hunk 2]
```diff
@@ -188,10 +182,40 @@ class CreateBakeTask implements RetryableTask {
...
       request.other().clear()
     }
+
     return request
   }
+
+  private static PackageType getCustomPackageType(StageExecution stage) {
+    PackageType type
+    if(stage.context.osType != "linux") {
+      type = PackageType.NUPKG
```
Comment (ovidiupopa07) [inline orca-bakery/src/main/groovy/com/netflix/spinnaker/orca/bakery/tasks/CreateBakeTask.groovy:205]: I would return instead the type. by doing this you don't need the `else` clause. you can directly do 
```
type = stage.context.packageType as PackageType
      if (type == null) {
        type = new OperatingSystem(stage.context.managedImage as String).getPackageType()
      }
```
without the else clause. For readability but it's just a personal preference.

**Review by ovidiupopa07 — CHANGES_REQUESTED** [review]
looks good but I would add unit tests for the new functionality. there are quite a few new if clauses added so I would test that the bake request it is what you expect

File: orca-bakery/src/main/groovy/com/netflix/spinnaker/orca/bakery/tasks/CreateBakeTask.groovy (hunk 2 above)
Comment (DanielaS12) [reply orca-bakery/src/main/groovy/com/netflix/spinnaker/orca/bakery/tasks/CreateBakeTask.groovy:205]: great catch, will do the changes

File: orca-bakery/src/main/groovy/com/netflix/spinnaker/orca/bakery/tasks/CreateBakeTask.groovy (hunk 1 above)
Comment (DanielaS12) [reply orca-bakery/src/main/groovy/com/netflix/spinnaker/orca/bakery/tasks/CreateBakeTask.groovy:185]: sure

**Review by ovidiupopa07 — APPROVED** [review]

**Comment by DanielaS12** [comment]
Closing in favor of https://github.com/spinnaker/orca/pull/4336

**Pull request closed without merging**